packages/element-plus/make-installer.ts

```diff
@@ -1,33 +1,14 @@
-import { setConfig } from '@element-plus/utils/config'
-import { localeContextKey, localeProviderMaker } from '@element-plus/hooks'
 import { version } from './version'
-
 import type { App, Plugin } from 'vue'
-import type { InstallOptions } from '@element-plus/utils/config'
-
-export const makeInstaller = (components: Plugin[] = []) => {
-  const apps: App[] = []
 
-  const install = (app: App, opts: InstallOptions) => {
-    const defaultInstallOpt: InstallOptions = {
-      size: '',
-      zIndex: 2000,
-    }
+const INSTALLED_KEY = Symbol('INSTALLED_KEY')
 
-    const option = Object.assign(defaultInstallOpt, opts)
-    if (apps.includes(app)) return
-    apps.push(app)
+export const makeInstaller = (components: Plugin[] = []) => {
+  const install = (app: App) => {
+    if (app[INSTALLED_KEY]) return
 
+    app[INSTALLED_KEY] = true
     components.forEach((c) => app.use(c))
-
-    if (option.locale) {
-      const localeProvides = localeProviderMaker(opts.locale)
-      app.provide(localeContextKey, localeProvides)
-    }
-
-    app.config.globalProperties.$ELEMENT = option
-    // app.provide() ? is this better? I think its not that flexible but worth implement
-    setConfig(option)
```

Comment:
We also need to provide the initial global configs to the application if the user is not using `config-provider` to wrap the application.

Reply:
IMO, we should let all user move to `config-provider`.

Reply:
The idea is that user should be able to use this library without `config-provider`, the default value we provided is for the underlying code to work correctly (to avoid potential bugs)

Reply:
I don't think we need to give two ways to set global configuration, please refer to `naive-ui` and `ant-design-vue`. And the configuration of config-provider is reactive.